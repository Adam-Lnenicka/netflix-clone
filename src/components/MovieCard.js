import React from "react";
import PropTypes from "prop-types";

const MovieCard = (prop) => {
  return (
    <div className="card" onClick={() => prop.function(prop.movie)}>
      <img
        className="card-image"
        src={prop.poster_path}
        alt={prop.title}
        onError={prop.broken}
      />
      <div className="card-details">
        <div>
          <h4>{prop.title}</h4>
          <p> {prop.genres}</p>
        </div>
        <button>{prop.release_date.slice(0, 4)}</button>
      </div>
    </div>
  );
};

MovieCard.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default MovieCard;
