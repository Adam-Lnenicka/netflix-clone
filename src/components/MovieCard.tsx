import React from "react";
import { useDispatch } from "react-redux";
import Shorten from "./Shorten";

import { removeMovieActionCreator } from "../store/actionCreators";
import { Link } from "react-router-dom";

interface MovieInterface {
  id: number;
  poster_path: string;
  title: string;
  broken: any;
  release_date: string;
  genres: string[];
}

const MovieCard = ({
  id,
  poster_path,
  title,
  broken,
  release_date,
  genres,
}: MovieInterface) => {
  const dispatch = useDispatch();

  return (
    <div className="card">
      <Link to={`/movie/${id}`}>
        <img
          className="card-image"
          src={poster_path}
          alt={title}
          onError={broken}
        />
        <div className="card-details">
          <div>
            <h4>{title}</h4>
            <p> {genres}</p>
          </div>
          <button>{Shorten(release_date)}</button>
          <button onClick={() => dispatch(removeMovieActionCreator(id))}>
            remove movie
          </button>{" "}
        </div>
      </Link>
    </div>
  );
};

export default MovieCard;
