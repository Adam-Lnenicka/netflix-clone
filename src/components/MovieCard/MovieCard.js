import React from "react";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import Shorten from "../Shorten";

import { removeMovieActionCreator } from "../../store/actionCreators";
import { Link } from "react-router-dom";
import { Card, CardDetails, Image } from "./styles/MovieCardStyles";

const MovieCard = ({
  id,
  poster_path,
  title,
  broken,
  release_date,
  genres,
}) => {
  const dispatch = useDispatch();

  return (
    <Card>
      <Link to={`/movie/${id}`}>
        <Image src={poster_path} alt={title} onError={broken} />
        <CardDetails>
          <div>
            <h4>{title}</h4>
            <p> {genres}</p>
          </div>
          <button>{Shorten(release_date)}</button>
          <button onClick={() => dispatch(removeMovieActionCreator(id))}>
            remove movie
          </button>{" "}
        </CardDetails>
      </Link>
    </Card>
  );
};

MovieCard.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default MovieCard;
