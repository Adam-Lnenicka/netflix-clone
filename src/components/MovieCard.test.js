import React from "react";
import { mount, shallow } from "enzyme";
import MovieCard from "./MovieCard";

describe("MovieCard", () => {
  it("snapshot", () => {
    const props = {
      title: "Captain Marvel",
      genres: ["Animation"],
      release_date: "2016",
      runtime: 108,
      description: "Movie description",
      poster_path: "https://",
      year: "2016",
      image: "movie image",
    };
    const component = mount(<MovieCard {...props} />);
    expect(component).toMatchSnapshot();
  });
});
