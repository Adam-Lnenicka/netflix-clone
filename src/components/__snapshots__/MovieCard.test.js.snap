// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MovieCard
    description="Movie description"
    genres={
      Array [
        "Animation",
      ]
    }
    poster_path="https://"
    release_date="2016"
    title="Captain Marvel"
    year="2016"
  />
   
</ContextProvider>
`;
